scripts/patch-nodes.sh

```diff
@@ -40,75 +53,66 @@ fi
+        $LOCAL_PATH/node.py --environment $ENV --role $ROLE --region $ORACLE_REGION --oracle --batch --inventory $RELEASE_PARAM >> $BASE_INVENTORY
+    done
+fi
 
-ANSIBLE_ROLES="${ANSIBLE_ROLES-"sshusers"}"
+LIVE_INVENTORY="./batch-${ROLE}-${ORACLE_REGION}-live.inventory"
 
-ANSIBLE_EXTRA_VARS="${ANSIBLE_EXTRA_VARS-""}"
-if [ -n "$ANSIBLE_EXTRA_VARS" ]; then
-  ANSIBLE_EXTRA_VARS="-e '$ANSIBLE_EXTRA_VARS'"
+SSH_FAILED_COUNT=0
+if [[ "$SKIP_SSH_CONFIRMATION" == "true" ]]; then
+    echo "## skipping ssh confirmation"
+    cp $BASE_INVENTORY $LIVE_INVENTORY
+else
+    while IFS='' read -r LINE || [ -n "$LINE" ]; do
+        IP=$(echo $LINE | awk '{ print $1 }')
+        echo "## confirming ssh liveness of $IP"
+        timeout 10 ssh -n -o StrictHostKeyChecking=no -F $LOCAL_PATH/../config/ssh.config $ANSIBLE_SSH_USER@$IP "uptime > /dev/null" && echo $LINE >> $LIVE_INVENTORY || SSH_FAILED_COUNT=$(($SSH_FAILED_COUNT+1))
+    done < "${BASE_INVENTORY}"
 fi
 
+echo "## slicing live inventory into batches"
 BATCH_SIZE=${BATCH_SIZE-"10"}
 
 [ -d ./.batch ] && rm -rf .batch
 mkdir .batch
-for ENV in $ENVIRONMENT_LIST; do
-    ANSIBLE_INVENTORY="./batch-${ROLE}-${ORACLE_REGION}-${ENV}.inventory"
-    split -l $BATCH_SIZE $ANSIBLE_INVENTORY ".batch/${ROLE}-${ORACLE_REGION}-${ENV}-"
-done
+split -l $BATCH_SIZE $LIVE_INVENTORY ".batch/${ROLE}-${ORACLE_REGION}-"
 
-FAILED_COUNT=0
 ANSIBLE_FAILURES=0
 
-set -x
-
-for ENV in $ENVIRONMENT_LIST; do
-    for BATCH_INVENTORY in .batch/${ROLE}-${ORACLE_REGION}-${ENV}*; do
-        echo "[tag_shard_role_$ROLE]" > ./batch.inventory
-        if [[ "$SKIP_SSH_CONFIRMATION" == "true" ]]; then
-            cat $BATCH_INVENTORY >> ./batch.inventory
-        else
-            for ip in $(cat $BATCH_INVENTORY | tail -n+1 | awk '{print $1}'); do
-                timeout 10 ssh -o StrictHostKeyChecking=no -F $LOCAL_PATH/../config/ssh.config $ANSIBLE_SSH_USER@$ip "uptime > /dev/null" && echo $ip >> ./batch.inventory || FAILED_COUNT=$(($FAILED_COUNT+1))
-            done
-        fi
-
-        LIVE_COUNT=$(cat ./batch.inventory | wc -l | awk '{print $1}')
-        if [[ $LIVE_COUNT -gt 1 ]]; then
-
-            ansible-playbook $ANSIBLE_PLAYBOOK \
-                -i ./batch.inventory \
-                -e "ansible_ssh_user=$ANSIBLE_SSH_USER hcv_environment=$ENV shard_role=$ROLE patch_ansible_roles=\"$ANSIBLE_ROLES\"" \
-                $ANSIBLE_EXTRA_VARS --vault-password-file .vault-password.txt \
-                --tags "$DEPLOY_TAGS"
-
-            if [[ $? -gt 0 ]]; then
-                echo "ERROR: Ansible batch failed for ${ENV}"
-                ANSIBLE_FAILURES=$(($ANSIBLE_FAILURES+1))
-            fi
-        else
-            echo "No live instances found in batch for ${ENV}, skipping"
-        fi
-    done
+echo "## starting $ANSIBLE_PLAYBOOK batch runs"
+for BATCH_INVENTORY in .batch/${ROLE}-${ORACLE_REGION}-*; do
+    echo "[tag_shard_role_$ROLE]" > ./batch.inventory
```

Comment:
This will add more than one label

Reply:
it's making a new file per batch with > instead of >>